tests/posix/xsi_threads_ext/src/main.c

```diff
@@ -0,0 +1,282 @@
+		zassert_ok(pthread_attr_getstack(&attr, &new_stackaddr, &new_stacksize));
+		zassert_equal(new_stackaddr, (void *)stack);
+		zassert_equal(new_stacksize, 2 * stacksize);
+		can_create_thread(&attr);
+	}
+}
+
+ZTEST(xsi_threads_ext, test_pthread_set_get_concurrency)
+{
+	/* EINVAL if the value specified by new_level is negative */
+	zassert_equal(EINVAL, pthread_setconcurrency(-42));
+
+	/*
+	 * Note: the special value 0 indicates the implementation will
+	 * maintain the concurrency level at its own discretion.
+	 *
+	 * pthread_getconcurrency() should return a value of 0 on init.
+	 */
+	zassert_equal(0, pthread_getconcurrency());
+
+	for (int i = 0; i <= CONFIG_MP_MAX_NUM_CPUS; ++i) {
+		zassert_ok(pthread_setconcurrency(i));
+		/* verify parameter is saved */
+		zassert_equal(i, pthread_getconcurrency());
+	}
+
+	/* EAGAIN if the a system resource to be exceeded */
+	zassert_equal(EAGAIN, pthread_setconcurrency(CONFIG_MP_MAX_NUM_CPUS + 1));
+}
+
+ZTEST(xsi_threads_ext, test_pthread_attr_getstacksize)
+{
+	size_t stacksize = BIOS_FOOD;
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_getstacksize(NULL, NULL), EINVAL);
+			zassert_equal(pthread_attr_getstacksize(NULL, &stacksize), EINVAL);
+			zassert_equal(pthread_attr_getstacksize(&uninit_attr, &stacksize), EINVAL);
+		}
+		zassert_equal(pthread_attr_getstacksize(&attr, NULL), EINVAL);
+	}
+
+	zassert_ok(pthread_attr_getstacksize(&attr, &stacksize));
+	zassert_not_equal(stacksize, BIOS_FOOD);
+}
+
+ZTEST(xsi_threads_ext, test_pthread_attr_setstacksize)
+{
+	size_t stacksize;
+	size_t new_stacksize;
+
+	/* valid size */
+	zassert_ok(pthread_attr_getstacksize(&attr, &stacksize));
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_setstacksize(NULL, 0), EINVAL);
+			zassert_equal(pthread_attr_setstacksize(NULL, stacksize), EINVAL);
+			zassert_equal(pthread_attr_setstacksize((pthread_attr_t *)&uninit_attr,
+								stacksize),
+				      EINVAL);
+		}
+		zassert_equal(pthread_attr_setstacksize(&attr, 0), EINVAL);
+	}
+
+	/* ensure we can spin up a thread with the default stack size */
+	can_create_thread(&attr);
+
+	/* set stack / addr to the current values of stack / addr */
+	zassert_ok(pthread_attr_setstacksize(&attr, stacksize));
+	/* ensure we can read back the values we just set */
+	zassert_ok(pthread_attr_getstacksize(&attr, &new_stacksize));
+	zassert_equal(new_stacksize, stacksize);
+	can_create_thread(&attr);
+
+	/* qemu_x86 seems to be unable to set thread stacks to be anything less than 4096 */
+	if (!IS_ENABLED(CONFIG_X86)) {
+		zassert_ok(pthread_attr_setstacksize(&attr, stacksize - 1));
+		/* ensure we can read back the values we just set */
+		zassert_ok(pthread_attr_getstacksize(&attr, &new_stacksize));
+		zassert_equal(new_stacksize, stacksize - 1);
+		can_create_thread(&attr);
+	}
+
+	if (IS_ENABLED(CONFIG_DYNAMIC_THREAD_ALLOC)) {
```

Comment:
Where is `CONFIG_DYNAMIC_THREAD_ALLOC` set in the `common` test suite and why doesn't it appear in this test suite?

Reply:
It should be in prj.conf now